src/libexec/zmrestoreblob

```diff
@@ -46,17 +105,50 @@ if ($orphaned_header_check) {
     for my $matched_blob (@matches) {
 
         my ( $filename, $dirs, $suffix ) = fileparse($matched_blob);
-        my $blob_fname = basename($filename);
+        my $blob_fname    = basename($filename);
+        my $org_blob_part = get_file_part($filename);
 
-        my $backup_path = $backup_dir . "/" . $blob_fname;
+        my @backup_zip_files = glob( $backup_dir . '/*' );
+        my $msg_match_flag   = 0;
+        foreach my $zip_file (@backup_zip_files) {
 
-# if blob exists in backup restore it in the origin blob path and display appropriate message.
-        if ( -f $backup_path ) {
-            print "[] INFO: Restoring $matched_blob from Backup\n";
-            copy( $backup_path, $matched_blob );
+            # Read a Zip file
+            my $backup_file      = '';
+            my $backup_blob_part = '';
+            my $zip              = Archive::Zip->new();
+            $zip->read($zip_file) == AZ_OK or die 'read error';
+            my @members = $zip->membersMatching('.*\.msg1');
+            foreach my $member (@members) {
+                $backup_file      = $member->fileName();
+                $backup_blob_part = get_file_part( fileparse($backup_file) );
+
+# if blob exists in backup, restore it in the origin blob path and display appropriate message.
+                if ( $org_blob_part eq $backup_blob_part ) {
+                    print "[] INFO: Restoring $matched_blob from Backup\n";
+                    $member->extractToFileNamed("$matched_blob");
```

Comment:
nit: remove the double quotes around $matched_blob - these should not be needed

Reply:
Done suggested changes and pushed, thanks